src/inet/UDPEndPoint.cpp

```diff
@@ -864,6 +864,10 @@ void UDPEndPoint::LwIPReceiveUDPMessage(void * arg, struct udp_pcb * pcb, struct
     System::LayerLwIP * lSystemLayer = static_cast<System::LayerLwIP *>(ep->Layer().SystemLayer());
     IPPacketInfo * pktInfo           = NULL;
     System::PacketBufferHandle buf   = System::PacketBufferHandle::Adopt(p);
+    if (!TryToFlattenPacket(buf))
+    {
+        return;
```

Comment:
ChipLogError at least please - I think this is the first place where this may return early and not call PostPacketBufferEvent.